.pipelines/shared-cluster.yml

```diff
@@ -0,0 +1,95 @@
+# No PR triggers as it runs on a schedule
+pr: none
+trigger: none
+
+schedules:
+- cron: 0 1 * * Sun
+  displayName: ⭕ Rebuild shared cluster Sun 1am (UTC)
+  always: true
+  branches:
+    include:
+    - master
+
+variables:
+- template: vars.yml
+  LOCATION: eastus
+  RESOURCEGROUP: v4-eastus
+  CLUSTER: aro-shared
+  VNET: dev-vnet
+
+jobs:
+- job: ⏬ Tear down shared cluster
+  pool:
+    vmImage: ARO-CI
+
+  steps:
+  - template: ./templates/template-setup-golang-env.yml
+    parameters:
+      gobin: ${{ variables.GOBIN }}
+      gopath: ${{ variables.GOPATH }}
+      goroot: ${{ variables.GOROOT }}
+      modulePath: ${{ variables.modulePath }}
+  - template: ./templates/template-setup-azure-tools.yml
+    parameters:
+      workingDirectory: ${{ variables.modulePath }}
+  - template: ./templates/template-az-cli-login.yml
+    parameters:
+      workingDirectory: ${{ variables.modulePath }}
+      azureDevOpsJSONSPN: $(aro-v4-ci-devops-spn)
+  - template: ./templates/template-az-aro-extension-prod.yml
+  - displayName: 🧹 Delete cluster
+    script: |
+      az aro delete --yes --resource-group "$RESOURCEGROUP" --name "$CLUSTER"
+
+- job: ⏫ Set up shared cluster
+  pool:
+    vmImage: ARO-CI
+
+  steps:
+  - template: ./templates/template-setup-golang-env.yml
+    parameters:
+      gobin: ${{ variables.GOBIN }}
+      gopath: ${{ variables.GOPATH }}
+      goroot: ${{ variables.GOROOT }}
+      modulePath: ${{ variables.modulePath }}
+  - template: ./templates/template-setup-azure-tools.yml
+    parameters:
+      workingDirectory: ${{ variables.modulePath }}
+  - template: ./templates/template-az-cli-login.yml
+    parameters:
+      workingDirectory: ${{ variables.modulePath }}
+      azureDevOpsJSONSPN: $(aro-v4-ci-devops-spn)
+  - template: ./templates/template-az-aro-extension-prod.yml
+  - displayName: 🚀 Create/update subnets
+    script: |
+      az network vnet subnet create \
+        --resource-group "$RESOURCEGROUP" \
+        --vnet-name "$VNET" \
+        --name "$CLUSTER-master" \
+        --address-prefixes 10.200.0.0/24 \
+        --service-endpoints Microsoft.ContainerRegistry
+      az network vnet subnet create \
+        --resource-group "$RESOURCEGROUP" \
+        --vnet-name "$VNET" \
+        --name "$CLUSTER-worker" \
+        --address-prefixes 10.201.0.0/24 \
+        --service-endpoints Microsoft.ContainerRegistry
+      az network vnet subnet update \
+        --name "$CLUSTER-master" \
+        --resource-group "$RESOURCEGROUP" \
+        --vnet-name "$VNET" \
+        --disable-private-link-service-network-policies true
+  - displayName: 🚀 Create cluster
+    script: |
+      set +x
+      export SECRET_SA_ACCOUNT_NAME=$(SECRET_SA_ACCOUNT_NAME)
+      make secrets
+      grep "^export USER_PULL_SECRET=" secrets/env | sed -e "s/.*'\([^']\+\)'.*/\1/" > pull-secret.json
```

Comment:
get the pull secret from a key vault via a vsts pipeline parameter - there's almost certainly one already there for the taking

Reply:
You mentioned here https://github.com/Azure/ARO-RP/pull/643#discussion_r421033514 there was no RH pull secret, and we already source the secrets in this way in some other pipelines, so there's actually no need for a new RH pipeline secret (which would be another point of failure in the event of rolling the secret in the future)

Reply:
643 was talking about a different secret.